src/miner.cpp

```diff
@@ -559,6 +541,86 @@ void static BitcoinMiner(CWallet *pwallet)
                     std::lock_guard<std::mutex> lock{m_cs};
                     return cancelSolver;
                 };
+
+
+#ifdef USE_TROMP_EQUIHASH
+            	//////////////////////////////////////////////////////////////////////////
+            	// TROMP EQ SOLVER START
+              	// I = the block header minus nonce and solution.
+                CEquihashInput I{*pblock};
+                CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+                ss << I;
+              	const char *header = (char *)&ss[0];
+              	unsigned int headerLen = ss.size();
+
+                // Nonce
+                // Nonce is u256 convert it to u32
+                const char *nonce = (const char*) pblock->nNonce.begin();
+
+              	// Create solver and initialize it with header and nonce.
+              	equi eq(1);
+              	eq.setnonce(header, headerLen, nonce, pblock->nNonce.size());
+
+                LogPrintf("Running TrompEquihash solver with nNonceInBlock=%s\n",
+                  		 pblock->nNonce.ToString());
+
+                // Intialization done, start algo driver.
```

Comment:
This section looks like the actual Equihash algorithm. It should be in a method somewhere in `src/equihash/tromp`, ideally using the same API as the current solver or the [Miner Challenge API](https://zcashminers.org/rules).